lib/cursor.js

```diff
@@ -366,6 +367,37 @@ Cursor.prototype._eachCb = function(err, data) {
   }
 }
 
+Cursor.prototype.asyncIterator = function() {
+  var self = this;
+  return {
+    next: function() {
+      var iter = this;
+      return self._next().then(value => {
+        return {
+          done: false,
+          value: value
+        };
+      }).error(function(error) {
+        if ((error.message !== 'You cannot retrieve data from a cursor that is closed.') &&
+            (error.message.match(/You cannot call `next` on a closed/) === null)) {
+          return iter.throw(error);
+        } else {
+          return iter.return();
```

Comment:
@mxstbr What other errors might occur?  why stop the iterations for only the above errors?

Reply:
Good question, I copied that from the other places where `self._next()` is called. Not sure why it's that specific, honestly!

Reply:
I suppose it makes sense, since any other errors you might want to ignore where as these don't make sense to continue.

This does raise a question though. With callbacks you get the error and can ignore or handle it.
In this case you wont get it. make it should be a part of the return `iter.return(err)`.

This also makes me really wonder what errors are possible... I'll ask around.